### PR TITLE
Tw 1154/member list isnot updated when i kicked someone

### PR DESCRIPTION
**Pull request by sherlockvn**
### Blockers:
https://github.com/linagora/matrix-dart-sdk/pull/43

### Demo:

https://github.com/linagora/twake-on-matrix/assets/43041967/f549b2f5-3571-4913-babb-c2de4ed631c5

### CODE REVIEW

File: lib/pages/chat_details/chat_details.dart [hunk 1]
```diff
@@ -176,7 +176,7 @@ class ChatDetailsController extends State<ChatDetails>
       future: () => room!.requestParticipants(),
     );
     if (participants.error == null) {
-      setState(() => members = participants.result);
+      setState(() => membersNotifier.value = participants.result);
```
Comment (imGok) [inline lib/pages/chat_details/chat_details.dart:179]: setState shouldn't be needed anymore if you use valueNotifier

File: lib/pages/chat_details/chat_details.dart [hunk 2]
```diff
@@ -221,6 +221,11 @@ class ChatDetailsController extends State<ChatDetails>
     );
   }
 
+  Future<void> onUpdateMembers() async {
+    final participants = await room!.requestParticipantsFromServer();
```
Comment (imGok) [inline lib/pages/chat_details/chat_details.dart:225]: To stay consistent 
```suggestion
    final members = await room!.requestParticipantsFromServer();
```

Comment (imGok) [inline lib/pages/chat_details/chat_details.dart:225]: To stay consistent 
```suggestion
    final members = await room!.requestParticipantsFromServer();
```

File: lib/pages/chat_details/chat_details_view.dart [hunk 3]
```diff
@@ -29,7 +29,8 @@ class ChatDetailsView extends StatelessWidget {
       );
     }
 
-    controller.members!.removeWhere((u) => u.membership == Membership.leave);
+    controller.membersNotifier.value!
```
Comment (imGok) [inline lib/pages/chat_details/chat_details_view.dart:32]: Are you sure it's safe ? 
` controller.membersNotifier.value!`

File: lib/pages/chat_details/participant_list_item.dart [hunk 4]
```diff
@@ -29,13 +35,20 @@ class ParticipantListItem extends StatelessWidget {
...
-        ),
+        onTap: () async {
+          final result = await showAdaptiveBottomSheet(
+            context: context,
+            builder: (c) => UserBottomSheet(
+              user: user,
+              outerContext: context,
+            ),
+          );
+          if (result is ChatMembersStatus) {
```
Comment (imGok) [inline lib/pages/chat_details/participant_list_item.dart:46]: unnecessary IMO 
`result == ChatMembersStatus.updated)` is enough

File: lib/pages/chat_details/chat_details_page_view/chat_details_members_page.dart [hunk 5]
```diff
@@ -7,60 +7,72 @@ import 'package:matrix/matrix.dart';
...
-              if (index == 0) {
-                return _listMemberInfoMobileAndTablet(context);
-              }
-              if (index - addMemberItemCount < members.length) {
-                return ParticipantListItem(members[index - addMemberItemCount]);
-              }
+    return ValueListenableBuilder(
+      valueListenable: membersNotifier,
+      builder: (context, members, child) {
+        members ??= [];
```
Comment (nqhhdev) [inline lib/pages/chat_details/chat_details_page_view/chat_details_members_page.dart:34]: Why not use `if`?

File: lib/pages/chat_details/chat_details_view.dart (hunk 3 above)
Comment (nqhhdev) [reply lib/pages/chat_details/chat_details_view.dart:32]: IMO: Move this func to controller, logic should not be handled directly in the view

File: lib/pages/chat_details/chat_details.dart (hunk 1 above)
Comment (sherlockvn) [reply lib/pages/chat_details/chat_details.dart:179]: done

File: lib/pages/chat_details/chat_details_page_view/chat_details_members_page.dart (hunk 5 above)
Comment (sherlockvn) [reply lib/pages/chat_details/chat_details_page_view/chat_details_members_page.dart:34]: same result but less code

File: lib/pages/chat_details/participant_list_item.dart (hunk 4 above)
Comment (sherlockvn) [reply lib/pages/chat_details/participant_list_item.dart:46]: it's okay, maybe we need to handle other cases of `ChatMembersStatus`

File: lib/pages/chat_details/chat_details_view.dart (hunk 3 above)
Comment (sherlockvn) [reply lib/pages/chat_details/chat_details_view.dart:32]: remove it, because it's not needed

**Review by nqhhdev — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**